ouroboros-network/src/Ouroboros/Network/PeerSelection/Churn.hs

```diff
@@ -497,4 +506,8 @@ peerChurnGovernor
 
     -- Replace 20% or at least one peer every churnInterval.
     decrease :: Int -> Int
-    decrease v = max 0 $ v  - max 1 (v `div` 5)
+    decrease = decreaseWithMin 1
+
+    -- Replace 20% or at least `u` or at least one peer every churnInterval.
+    decreaseWithMin :: Int -> Int -> Int
+    decreaseWithMin u v =  max 0 $ v - max u (max 1 (v `div` 5))
```

Comment:
If `20%` of active peers is more than all warm peers, we'll churn all warm peers plus some established.  A sane configuration should be such that `0.2 * active <= (established - active)`, an example of insane :wink: configuration: `55` active peers, `65` established, each turn we'd churn `11` established peers.

Does this deserve a warning?

Reply:
We would actually churn 10 established peers. since `max 0` protects us from churning more than the number of established - active.

So every churn we'll replace all established peers. I am not sure it would warrant a warning